src/planscape/collaboration/permissions.py

```diff
@@ -0,0 +1,99 @@
+from collaboration.utils import check_for_permission, is_creator
+from planning.models import PlanningArea, Scenario
+from django.contrib.auth.models import User
+
+
+class CheckPermissionMixin:
+    @staticmethod
+    def can_view():
+        raise NotImplementedError("Subclasses must implement can_add.")
+
+    @staticmethod
+    def can_add():
+        raise NotImplementedError("Subclasses must implement can_add.")
+
+    @staticmethod
+    def can_change():
+        raise NotImplementedError("Subclasses must implement can_add.")
+
+    @staticmethod
+    def can_remove():
+        raise NotImplementedError("Subclasses must implement can_remove.")
+
+
+class PlanningAreaPermission(CheckPermissionMixin):
+    @staticmethod
+    def can_view(user: User, planning_area: PlanningArea):
+        if is_creator(user, planning_area):
```

Comment:
one more nit as a possible refactor idea: I'm wondering if we could also make `is_creator` a static method of CheckPermission mixin, perhaps along with `check_for_permission`. 
For the latter, subclasses could also set the object type as part of the class, rather than have to check it each time.